botocore/endpoint.py

```diff
@@ -171,7 +171,12 @@ def _get_response(self, request, operation_model, attempts):
                 'url': request.url,
                 'body': request.body
             })
-            http_response = self._send(request)
+            service_id = operation_model.service_model.service_id.hyphenize()
+            event_name = 'send-request.%s.%s' % (service_id, operation_model.name)
```

Comment:
How about calling this `before-send` to be more descriptive and more consistent with our other event names